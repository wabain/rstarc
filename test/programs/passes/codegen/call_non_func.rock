Hidden Condition takes X, Y
    Give back X with Y
    (End function)

Foo takes X, Y
    Say X
    Say Y
    (End function)

If Hidden Condition taking 1, 2 is greater than 7
    Put Foo into Bar
    (End if)

Say "Calling non-function..."
Put Bar taking 1 into Result
Say Result
