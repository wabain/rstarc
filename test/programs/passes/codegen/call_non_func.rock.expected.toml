[tests]
enabled = ["ir", "run"]

[ir]
returncode = "0"
stdout = """
main:
  Bar := store mysterious
  Foo := store mysterious
  Hidden Condition := store mysterious
  Result := store mysterious
  Hidden Condition := store Function(1)
  Foo := store Function(2)
  t0 := call Hidden Condition, [1, 2]
  t1 := gt t0, 7
  jumpif t1, .then0, .else1
.then0:
  Bar := store Foo
  jump .else1
.else1:
  say "Calling non-function..."
  Result := call Bar, [1]
  say Result
  return-default

;
; Function Hidden Condition takes X, Y
; Scope is 1
;
f1_Hidden_Condition:
  X := load-arg 0
  Y := load-arg 1
  t0 := add X, Y
  return t0

;
; Function Foo takes X, Y
; Scope is 2
;
f2_Foo:
  X := load-arg 0
  Y := load-arg 1
  say X
  say Y
  return-default
"""

[run]
returncode = "1"
stdout = """
Calling non-function...
"""
stderr = """
error: Cannot call value 'mysterious'
"""
